.github/workflows/publish.yml

```diff
@@ -2,23 +2,38 @@
 name: Publish
 on:  # yamllint disable-line rule:truthy
   push:
-    branches:
-      - main
+    # branches:
+    #   - main
 jobs:
   ci:
     name: Run ci
     uses: ./.github/workflows/ci.yml
 
+  build:
+    name: Build Alertmanager for all architectures
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        thread: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]
+    needs: ci
+    steps:
+      - uses: actions/checkout@v3
+      - uses: prometheus/promci@3cb0c3871f223bd5ce1226995bd52ffb314798b6 # v0.1.0
+      - uses: ./.github/promci/actions/build
+        with:
+          parallelism: 12
+          thread: ${{ matrix.thread }}
   publish_main:
     name: Publish main branch artefacts
     runs-on: ubuntu-latest
-    needs: ci
+    needs: build
     steps:
       - uses: actions/checkout@v3
-      - uses: prometheus/promci@v0.0.2
+      - uses: prometheus/promci@3cb0c3871f223bd5ce1226995bd52ffb314798b6 # v0.1.0
       - uses: ./.github/promci/actions/publish_main
         with:
           docker_hub_login: ${{ secrets.docker_hub_login }}
           docker_hub_password: ${{ secrets.docker_hub_password }}
+          quay_io_organization: ${{ secrets.quay_io_organization }}
           quay_io_login: ${{ secrets.quay_io_login }}
           quay_io_password: ${{ secrets.quay_io_password }}
```

Comment:
None of these secrets are defined. At which point would be ready to test them? How did you approach this in Prometheus?

Reply:
Yeah I'm looking into that, not sure myself at this point. In my clone I defined those secrets but it still fails, there is something I don't understand with multi arch builds.

[If those secrets don't exist the publish action is just a no-op](https://github.com/prometheus/promci/blob/3cb0c3871f223bd5ce1226995bd52ffb314798b6/actions/publish_main/action.yml#L31-L38). so we can leave this for now and once we understand what's going on, defining the secrets is all that is needed to publish from the action.

Reply:
Great to know!

Reply:
they should be defined at the prometheus org level